<launch>
    <arg name="veh" doc="Name of vehicle. ex: megaman"/>
    <arg name="pkg_name" value="apriltag_behaviors"/>
    <arg name="node_name" default="apriltag_behaviors_node"/>
    <arg name="config" default="baseline" doc="Specify a config."/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex: duckiebot."/>

    <!-- Subscription -->
    <!-- "detections": duckietown_msgs/AprilTagDetectionArray. Tag detections. -->

    <!-- Publication -->

    <group ns="$(arg veh)">
        <remap from="apriltag_behaviors_node/detections" to="apriltag_detector_node/detections"/>
<!--        <remap from="apriltag_behaviors_node/joystick_override" to="joy_mapper_node/joystick_override"/>-->
        <remap from="apriltag_behaviors_node/estop" to="wheels_driver_node/emergency_stop"/>
        <remap from="apriltag_behaviors_node/set_custom_pattern" to="led_emitter_node/set_custom_pattern"/>

        <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen">
            <param name="veh" value="$(arg veh)" />
            <rosparam command="load"
                file="$(find apriltag_behaviors)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
    </group>
</launch>
